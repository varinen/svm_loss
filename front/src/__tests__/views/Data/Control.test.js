import React from "react";
import DataControl from "views/SVMLoss/Demo/Data/Control";
import {createShallow} from '@material-ui/core/test-utils';

describe('Data control renders correct grid', () => {
    let component, shallow;
    const mockFetchData = jest.fn();
    const props = {
        data: [[1, 2, 3], [4, 5, 6], [7, 8, 9]],
        fetchData: mockFetchData
    };
    beforeEach(() => {
        shallow = createShallow({dive: true});
        component = shallow(<DataControl {...props}/>);
    });

    it('renders correctly', () => {
        expect(component).toMatchSnapshot();
    });

    it('Generates a correct number of table rows elements', () => {
        expect(component.find('[data-table-data-row="1"]').length).toEqual(props.data.length);
    });


    it('Click on randomize button causes the fetchData be called', () => {
        const b0 = 10;
        component.find('#rand-data')
            .simulate('click');
        expect(mockFetchData).toHaveBeenCalledWith(1);
    });
});
