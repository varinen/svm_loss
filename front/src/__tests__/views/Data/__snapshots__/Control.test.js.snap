// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Data control renders correct grid renders correctly 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Table))
    aria-label="simple table"
  >
    <WithStyles(ForwardRef(TableHead))>
      <WithStyles(ForwardRef(TableRow))>
        <WithStyles(ForwardRef(TableCell))
          align="right"
          className="DataControl-psm-1"
        >
          X[0]
        </WithStyles(ForwardRef(TableCell))>
        <WithStyles(ForwardRef(TableCell))
          align="right"
          className="DataControl-psm-1"
        >
          X[1]
        </WithStyles(ForwardRef(TableCell))>
        <WithStyles(ForwardRef(TableCell))
          align="right"
          className="DataControl-borderRight-22 DataControl-psm-1"
        >
          y
        </WithStyles(ForwardRef(TableCell))>
        <WithStyles(ForwardRef(TableCell))
          align="right"
          className="DataControl-psm-1"
        >
          s[0]
        </WithStyles(ForwardRef(TableCell))>
        <WithStyles(ForwardRef(TableCell))
          align="right"
          className="DataControl-psm-1"
        >
          s[1]
        </WithStyles(ForwardRef(TableCell))>
        <WithStyles(ForwardRef(TableCell))
          align="right"
          className="DataControl-borderRight-22 DataControl-psm-1"
        >
          s[3]
        </WithStyles(ForwardRef(TableCell))>
        <WithStyles(ForwardRef(TableCell))
          align="right"
          className="DataControl-psm-1"
        >
          L
        </WithStyles(ForwardRef(TableCell))>
      </WithStyles(ForwardRef(TableRow))>
    </WithStyles(ForwardRef(TableHead))>
    <WithStyles(ForwardRef(TableBody))>
      <WithStyles(ForwardRef(TableRow))
        data-table-data-row="1"
        key="data-row0"
      >
        <WithStyles(ForwardRef(TableCell))
          align="right"
          className="DataControl-psm-1"
          data-table-data-cell="1"
          key="data-cell-0-0"
        >
          1.00
        </WithStyles(ForwardRef(TableCell))>
        <WithStyles(ForwardRef(TableCell))
          align="right"
          className="DataControl-psm-1"
          data-table-data-cell="1"
          key="data-cell-0-1"
        >
          2.00
        </WithStyles(ForwardRef(TableCell))>
        <WithStyles(ForwardRef(TableCell))
          align="right"
          className="DataControl-borderRight-22 DataControl-psm-1"
          data-table-data-cell="1"
          key="data-cell-0-2"
        >
          3.00
        </WithStyles(ForwardRef(TableCell))>
        <WithStyles(ForwardRef(TableCell))
          align="right"
          className="DataControl-psm-1"
          data-table-data-cell="1"
          key="data-cell-0-3"
        >
          0.00
        </WithStyles(ForwardRef(TableCell))>
        <WithStyles(ForwardRef(TableCell))
          align="right"
          className="DataControl-psm-1"
          data-table-data-cell="1"
          key="data-cell-0-4"
        >
          0.00
        </WithStyles(ForwardRef(TableCell))>
        <WithStyles(ForwardRef(TableCell))
          align="right"
          className="DataControl-borderRight-22 DataControl-psm-1"
          data-table-data-cell="1"
          key="data-cell-0-5"
        >
          0.00
        </WithStyles(ForwardRef(TableCell))>
        <WithStyles(ForwardRef(TableCell))
          align="right"
          className="DataControl-psm-1"
          data-table-data-cell="1"
          key="data-cell-0-6"
        >
          0.00
        </WithStyles(ForwardRef(TableCell))>
      </WithStyles(ForwardRef(TableRow))>
      <WithStyles(ForwardRef(TableRow))
        data-table-data-row="1"
        key="data-row1"
      >
        <WithStyles(ForwardRef(TableCell))
          align="right"
          className="DataControl-psm-1"
          data-table-data-cell="1"
          key="data-cell-1-0"
        >
          4.00
        </WithStyles(ForwardRef(TableCell))>
        <WithStyles(ForwardRef(TableCell))
          align="right"
          className="DataControl-psm-1"
          data-table-data-cell="1"
          key="data-cell-1-1"
        >
          5.00
        </WithStyles(ForwardRef(TableCell))>
        <WithStyles(ForwardRef(TableCell))
          align="right"
          className="DataControl-borderRight-22 DataControl-psm-1"
          data-table-data-cell="1"
          key="data-cell-1-2"
        >
          6.00
        </WithStyles(ForwardRef(TableCell))>
        <WithStyles(ForwardRef(TableCell))
          align="right"
          className="DataControl-psm-1"
          data-table-data-cell="1"
          key="data-cell-1-3"
        >
          0.00
        </WithStyles(ForwardRef(TableCell))>
        <WithStyles(ForwardRef(TableCell))
          align="right"
          className="DataControl-psm-1"
          data-table-data-cell="1"
          key="data-cell-1-4"
        >
          0.00
        </WithStyles(ForwardRef(TableCell))>
        <WithStyles(ForwardRef(TableCell))
          align="right"
          className="DataControl-borderRight-22 DataControl-psm-1"
          data-table-data-cell="1"
          key="data-cell-1-5"
        >
          0.00
        </WithStyles(ForwardRef(TableCell))>
        <WithStyles(ForwardRef(TableCell))
          align="right"
          className="DataControl-psm-1"
          data-table-data-cell="1"
          key="data-cell-1-6"
        >
          0.00
        </WithStyles(ForwardRef(TableCell))>
      </WithStyles(ForwardRef(TableRow))>
      <WithStyles(ForwardRef(TableRow))
        data-table-data-row="1"
        key="data-row2"
      >
        <WithStyles(ForwardRef(TableCell))
          align="right"
          className="DataControl-psm-1"
          data-table-data-cell="1"
          key="data-cell-2-0"
        >
          7.00
        </WithStyles(ForwardRef(TableCell))>
        <WithStyles(ForwardRef(TableCell))
          align="right"
          className="DataControl-psm-1"
          data-table-data-cell="1"
          key="data-cell-2-1"
        >
          8.00
        </WithStyles(ForwardRef(TableCell))>
        <WithStyles(ForwardRef(TableCell))
          align="right"
          className="DataControl-borderRight-22 DataControl-psm-1"
          data-table-data-cell="1"
          key="data-cell-2-2"
        >
          9.00
        </WithStyles(ForwardRef(TableCell))>
        <WithStyles(ForwardRef(TableCell))
          align="right"
          className="DataControl-psm-1"
          data-table-data-cell="1"
          key="data-cell-2-3"
        >
          0.00
        </WithStyles(ForwardRef(TableCell))>
        <WithStyles(ForwardRef(TableCell))
          align="right"
          className="DataControl-psm-1"
          data-table-data-cell="1"
          key="data-cell-2-4"
        >
          0.00
        </WithStyles(ForwardRef(TableCell))>
        <WithStyles(ForwardRef(TableCell))
          align="right"
          className="DataControl-borderRight-22 DataControl-psm-1"
          data-table-data-cell="1"
          key="data-cell-2-5"
        >
          0.00
        </WithStyles(ForwardRef(TableCell))>
        <WithStyles(ForwardRef(TableCell))
          align="right"
          className="DataControl-psm-1"
          data-table-data-cell="1"
          key="data-cell-2-6"
        >
          0.00
        </WithStyles(ForwardRef(TableCell))>
      </WithStyles(ForwardRef(TableRow))>
    </WithStyles(ForwardRef(TableBody))>
  </WithStyles(ForwardRef(Table))>
  <WithStyles(ForwardRef(Divider))
    className="DataControl-mt1-6 DataControl-mb1-10"
  />
  <ForwardRef
    color="primary"
    id="rand-data"
    onClick={[Function]}
  >
    Randomize
  </ForwardRef>
</Fragment>
`;
